### ISSUE TITLE
YugabyteDB tserver crashes immediately on startup

### ISSUE THREAD

**Issue by dobesv** [yugabyte/yugabyte-db#4026]
I followed the basic Linux instructions on the web site, but the `tserver` just crashes on startup.

```
$ wget https://downloads.yugabyte.com/yugabyte-2.1.2.0-linux.tar.gz
...
2020-03-20 14:24:52 (47.1 MB/s) - ‘yugabyte-2.1.2.0-linux.tar.gz’ saved [488471742/488471742]
$ tar -zxvf yugabyte-2.1.2.0-linux.tar.gz 
...
$ cd yugabyte-2.1.2.0/
$ ./bin/post_install.sh 
+ /home/ubuntu/yugabyte-2.1.2.0/bin/patchelf --set-interpreter /home/ubuntu/yugabyte-2.1.2.0/lib/ld.so ldb
+ /home/ubuntu/yugabyte-2.1.2.0/bin/patchelf --set-rpath /home/ubuntu/yugabyte-2.1.2.0/lib/yb:/home/ubuntu/yugabyte-2.1.2.0/lib/yb-thirdparty:/home/ubuntu/yugabyte-2.1.2.0/linuxbrew/lib ldb
...
$ ./bin/yb-ctl create
$ ./bin/yb-ctl create
Creating cluster.
Waiting for cluster to be ready.
Viewing file /home/ubuntu/yugabyte-data/node-1/disk-1/tserver.err:
In YugabyteDB, setting LC_COLLATE to C and all other locale settings to en_US.UTF-8 by default. Locale support will be enhanced as part of addressing https://github.com/yugabyte/yugabyte-db/issues/15572020-03-20 21:26:42.717 UTC [17258] LOG:  YugaByte is ENABLED in PostgreSQL. Transactions are enabled.
2020-03-20 21:26:42.720 UTC [17258] LOG:  listening on IPv4 address "127.0.0.1", port 5433
2020-03-20 21:26:42.729 UTC [17258] LOG:  redirecting log output to logging collector process
2020-03-20 21:26:42.729 UTC [17258] HINT:  Future log output will appear in directory "/home/ubuntu/yugabyte-data/node-1/disk-1/yb-data/tserver/logs".
*** Aborted at 1584739603 (unix time) try "date -d @1584739603" if you are using GNU date ***
PC: @     0x7f87e6497939 std::_Rb_tree<>::_M_erase()
*** SIGSEGV (@0x3e) received by PID 17176 (TID 0x7f87c9c90700) from PID 62; stack trace: ***
    @     0x7f87d6c10ba0 (unknown)
    @     0x7f87e6497938  std::_Rb_tree<>::_M_erase()
    @     0x7f87dc3bd40d  yb::Subprocess::~Subprocess()
    @     0x7f87de7f5f03  yb::pgwrapper::PgSupervisor::RunThread()
    @     0x7f87dc3cafae  yb::Thread::SuperviseThread()
    @     0x7f87d6c08694 start_thread
    @     0x7f87d634541d __clone
    @                0x0 (unknown)
Viewing file /tmp/tmpe4Rmel:
2020-03-20 14:26:41,770 INFO: Starting master-1 with:
/home/ubuntu/yugabyte-2.1.2.0/bin/yb-master --fs_data_dirs "/home/ubuntu/yugabyte-data/node-1/disk-1" --webserver_interface 127.0.0.1 --rpc_bind_addresses 127.0.0.1 --v 0 --version_file_json_path=/home/ubuntu/yugabyte-2.1.2.0 --webserver_doc_root "/home/ubuntu/yugabyte-2.1.2.0/www" --replication_factor=1 --yb_num_shards_per_tserver 2 --ysql_num_shards_per_tserver=2 --master_addresses 127.0.0.1:7100 --enable_ysql=true >"/home/ubuntu/yugabyte-data/node-1/disk-1/master.out" 2>"/home/ubuntu/yugabyte-data/node-1/disk-1/master.err" &
2020-03-20 14:26:41,781 INFO: Starting tserver-1 with:
/home/ubuntu/yugabyte-2.1.2.0/bin/yb-tserver --fs_data_dirs "/home/ubuntu/yugabyte-data/node-1/disk-1" --webserver_interface 127.0.0.1 --rpc_bind_addresses 127.0.0.1 --v 0 --version_file_json_path=/home/ubuntu/yugabyte-2.1.2.0 --webserver_doc_root "/home/ubuntu/yugabyte-2.1.2.0/www" --tserver_master_addrs=127.0.0.1:7100 --yb_num_shards_per_tserver=2 --redis_proxy_bind_address=127.0.0.1:6379 --cql_proxy_bind_address=127.0.0.1:9042 --local_ip_for_outbound_sockets=127.0.0.1 --use_cassandra_authentication=false --ysql_num_shards_per_tserver=2 --enable_ysql=true --pgsql_proxy_bind_address=127.0.0.1:5433 >"/home/ubuntu/yugabyte-data/node-1/disk-1/tserver.out" 2>"/home/ubuntu/yugabyte-data/node-1/disk-1/tserver.err" &
2020-03-20 14:26:41,783 INFO: Waiting for master and tserver processes to come up.
2020-03-20 14:26:41,802 INFO: Waiting for master leader election and tablet server registration.
2020-03-20 14:26:44,137 INFO: PIDs found: {'tserver': [None], 'master': [17173]}
2020-03-20 14:26:44,137 ERROR: At least one master or tserver process is down.
^^^ Encountered errors ^^^
```

Also tried wipe_restart and got a different `SIGSEGV`:

```
$ ./bin/yb-ctl wipe_restart
Destroying cluster.
Creating cluster.
Waiting for cluster to be ready.
Viewing file /home/ubuntu/yugabyte-data/node-1/disk-1/tserver.err:
In YugabyteDB, setting LC_COLLATE to C and all other locale settings to en_US.UTF-8 by default. Locale support will be enhanced as part of addressing https://github.com/yugabyte/yugabyte-db/issues/15572020-03-20 21:27:55.255 UTC [17402] LOG:  YugaByte is ENABLED in PostgreSQL. Transactions are enabled.
2020-03-20 21:27:55.259 UTC [17402] LOG:  listening on IPv4 address "127.0.0.1", port 5433
2020-03-20 21:27:55.270 UTC [17402] LOG:  redirecting log output to logging collector process
2020-03-20 21:27:55.270 UTC [17402] HINT:  Future log output will appear in directory "/home/ubuntu/yugabyte-data/node-1/disk-1/yb-data/tserver/logs".
F0320 21:27:56.279376 17403 pg_wrapper.cc:292] Check failed: _s.ok() Bad status: IO error (yb/yql/pgwrapper/pg_wrapper.cc:121): Failed to write ysql pg configuration: : Failed to read default postgres configuration '%s': errno=i�330f�/postgresql.conf: 2
Fatal failure details written to /home/ubuntu/yugabyte-data/node-1/disk-1/yb-data/tserver/logs/yb-tserver.FATAL.details.2020-03-20T21_27_56.pid17320.txt
F20200320 21:27:56 ../../../../../../src/yb/yql/pgwrapper/pg_wrapper.cc:292] Check failed: _s.ok() Bad status: IO error (yb/yql/pgwrapper/pg_wrapper.cc:121): Failed to write ysql pg configuration: : Failed to read default postgres configuration '%s': errno=i�330f�/postgresql.conf: 2
    @     0x7fad6b1aa74c  yb::LogFatalHandlerSink::send()
    @     0x7fad6a396336  google::LogMessage::SendToLog()
    @     0x7fad6a39379a  google::LogMessage::Flush()
    @     0x7fad6a396869  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fad6d661c60  yb::pgwrapper::PgWrapper::Start()
    @     0x7fad6d66270c  yb::pgwrapper::PgSupervisor::StartServerUnlocked()
    @     0x7fad6d662c94  yb::pgwrapper::PgSupervisor::RunThread()
    @     0x7fad6b237faf  yb::Thread::SuperviseThread()
    @     0x7fad65a75694  start_thread
    @     0x7fad651b241d  __clone
    @              (nil)  (unknown)

*** Check failure stack trace: ***
    @     0x7fad6b1a8b31  yb::(anonymous namespace)::DumpStackTraceAndExit()
    @     0x7fad6a393c4d  google::LogMessage::Fail()
    @     0x7fad6a395dbd  google::LogMessage::SendToLog()
    @     0x7fad6a39379a  google::LogMessage::Flush()
    @     0x7fad6a396869  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fad6d661c60  yb::pgwrapper::PgWrapper::Start()
    @     0x7fad6d66270c  yb::pgwrapper::PgSupervisor::StartServerUnlocked()
    @     0x7fad6d662c94  yb::pgwrapper::PgSupervisor::RunThread()
    @     0x7fad6b237faf  yb::Thread::SuperviseThread()
    @     0x7fad65a75694  start_thread
    @     0x7fad651b241d  __clone
    @              (nil)  (unknown)
Viewing file /tmp/tmpfefy6Q:
2020-03-20 14:27:53,885 INFO: Stopping Server master-1 PID=17173
2020-03-20 14:27:53,885 INFO: Waiting for Server master-1 PID=17173 to stop...
2020-03-20 14:27:54,396 INFO: Server tserver-1 already stopped
2020-03-20 14:27:54,396 INFO: Stopping Postmaster for tserver-1 PID=17258
2020-03-20 14:27:54,396 INFO: Postmaster for tserver-1 does not exist; nothing to stop
2020-03-20 14:27:54,396 INFO: Waiting for Postmaster for tserver-1 PID=17258 to stop...
2020-03-20 14:27:54,396 INFO: Removing base directory: /home/ubuntu/yugabyte-data
2020-03-20 14:27:54,429 INFO: Starting master-1 with:
/home/ubuntu/yugabyte-2.1.2.0/bin/yb-master --fs_data_dirs "/home/ubuntu/yugabyte-data/node-1/disk-1" --webserver_interface 127.0.0.1 --rpc_bind_addresses 127.0.0.1 --v 0 --version_file_json_path=/home/ubuntu/yugabyte-2.1.2.0 --webserver_doc_root "/home/ubuntu/yugabyte-2.1.2.0/www" --replication_factor=1 --yb_num_shards_per_tserver 2 --ysql_num_shards_per_tserver=2 --master_addresses 127.0.0.1:7100 --enable_ysql=true >"/home/ubuntu/yugabyte-data/node-1/disk-1/master.out" 2>"/home/ubuntu/yugabyte-data/node-1/disk-1/master.err" &
2020-03-20 14:27:54,441 INFO: Starting tserver-1 with:
/home/ubuntu/yugabyte-2.1.2.0/bin/yb-tserver --fs_data_dirs "/home/ubuntu/yugabyte-data/node-1/disk-1" --webserver_interface 127.0.0.1 --rpc_bind_addresses 127.0.0.1 --v 0 --version_file_json_path=/home/ubuntu/yugabyte-2.1.2.0 --webserver_doc_root "/home/ubuntu/yugabyte-2.1.2.0/www" --tserver_master_addrs=127.0.0.1:7100 --yb_num_shards_per_tserver=2 --redis_proxy_bind_address=127.0.0.1:6379 --cql_proxy_bind_address=127.0.0.1:9042 --local_ip_for_outbound_sockets=127.0.0.1 --use_cassandra_authentication=false --ysql_num_shards_per_tserver=2 --enable_ysql=true --pgsql_proxy_bind_address=127.0.0.1:5433 >"/home/ubuntu/yugabyte-data/node-1/disk-1/tserver.out" 2>"/home/ubuntu/yugabyte-data/node-1/disk-1/tserver.err" &
2020-03-20 14:27:54,442 INFO: Waiting for master and tserver processes to come up.
2020-03-20 14:27:54,464 INFO: Waiting for master leader election and tablet server registration.
2020-03-20 14:27:56,840 INFO: PIDs found: {'tserver': [None], 'master': [17317]}
2020-03-20 14:27:56,841 ERROR: At least one master or tserver process is down.
```

**Comment by dobesv** [comment]
```
$ uname -a
Linux 5.0.0-29-generic #31~18.04.1-Ubuntu SMP Thu Sep 12 18:29:21 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by bmatican** [comment]
Hey @dobesv . Thanks for trying out yugabyte!

We've recently introduced a change to how we report errors from main (adf5b66df1dca71861bdefd21bf02417a04e9e40), but we have not yet released it yet.

Given that the tserver died, it is likely the root cause is that something else is binding to one of the ports we try to bind on startup. Do you mind checking in your tserver INFO log file (`/home/ubuntu/yugabyte-data/node-1/disk-1/yb-data/tserver/logs/yb-tserver.INFO`) for a message like `Address already in use`?

Alternatively, if I'm wrong about the root cause, do you mind just sharing the complete `yb-tserver.INFO` file in the issue?

**Comment by dobesv** [comment]
Aha, good hint.  When I tried to run using `docker-compose` it complained clearly about port `6379` not being available, so I stopped my local `redis-server`, which made it happy.  Now when I try to run the server outside docker, it works.  If I start `redis-server` and try to run the cluster I get a similar error as reported above.

So the problem was port 6379 not being available.  Thanks for the speedy response!

Sounds like you are already working on this so I'll close the issue.
